### ISSUE TITLE
Property 'containers' does not exist on type 'PeaksOptions'

### ISSUE THREAD

**Issue by cathino** [suterma/vue-peaks#7]
Hello, there is an error in my project when I execute ***pnpm build**, and this project made by ts + vite,

```
> vite build

vite v4.4.9 building for production...
transforming (423) node_modules\.pnpm\element-plus@2.3.12_vue@3.3.4\node_modules\element-plus\es\components\menu\src\use-menu.mjsnode_modules/vue-peaks/src/components/AudioPeaks.vue:130:20 - error TS2339: Property 'containers' does not exist on type 'PeaksOptions'.
  Property 'containers' does not exist on type 'OptionalOptions & RemoteWaveformDataOptions & Without<LocalWaveformDataOptions> & Without<WebAudioOptions> & ContainerOptions'.

130     (props.options.containers = {
                       ~~~~~~~~~~

node_modules/vue-peaks/src/components/AudioPeaks.vue:140:7 - error TS2322: Type 'PeaksOptions | { containers: { overview: HTMLDivElement | undefined; zoomview: HTMLDivElement | undefined; }; mediaElement: HTMLMediaElement | undefined; webAudio: { ...; }; zoomLevels: number[]; }' is not assignable to type 'PeaksOptions'.
  Type '{ containers: { overview: HTMLDivElement | undefined; zoomview: HTMLDivElement | undefined; }; mediaElement: HTMLMediaElement | undefined; webAudio: { ...; }; zoomLevels: number[]; }' is not assignable to type 'PeaksOptions'.
    Object literal may only specify known properties, and 'containers' does not exist in type 'OptionalOptions & Without<RemoteWaveformDataOptions> & Without<LocalWaveformDataOptions> & WebAudioOptions & ContainerOptions'.

140       containers: {
          ~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/vue-peaks/src/components/AudioPeaks.vue:130

 ELIFECYCLE  Command failed with exit code 2.
transforming (523) node_modules\.pnpm\@floating-ui+dom@1.5.1\node_modules\@floating-ui\dom\dist\floating-ui.dom.mjsERROR: "type-check" exited with 1.
 ELIFECYCLE  Command failed with exit code 1.
```

My vue code:
```vue
    <audio ref="audioRef" :src="audioUrl" @canplay="onCanplay" id='externalMediaElement'></audio>

    <AudioPeaks mediaElementId='externalMediaElement' ></AudioPeaks>
```

Is there a wrong type defined in PeaksOptions?

**Comment by cathino** [comment]
My peaks.js version is 3.0.0, and after I modified the code in **src/components/AudioPeaks.vue** like :

```js
  if (props.options) {
      // Comment the following 6 lines:
     // (
      // props.options.containers = {
      //   overview: overviewElement,
      //   zoomview: zoomviewElement,
      // }   
    // ),
      (props.options.mediaElement = mediaElement);

      // Add the following 6 lines:
      if(props.options.overview){
        props.options.overview.container=overviewElement
      }
      if(props.options.zoomview){
        props.options.zoomview.container=zoomviewElement
      }
  }

  const options: PeaksOptions = props.options
    ? props.options
    : {
      // Comment the following 4 lines:
      // containers: {
      //   overview: overviewElement,
      //   zoomview: zoomviewElement,
      // },
     // Add the following 2 lines:
      overview:{container:overviewElement},
      zoomview:{container:zoomviewElement},

      mediaElement: mediaElement,
      webAudio: {
        audioContext: new AudioContext(),
      },
      zoomLevels: [256, 512, 1024, 2048, 4096],
    };
```

It works.

**Comment by chrisn** [comment]
Yes, Peaks.js v3.0.0 has some breaking API changes, the [migration guide](https://github.com/bbc/peaks.js/blob/master/doc/migration-guide.md) explains how to update your code.

**Comment by suterma** [comment]
Hi @newproplus and @chrisn and thanks for reporting and commenting. I will look into it.

**Comment by suterma** [comment]
I can confirm that vue-peaks does not work with peaks.js v3 (however it currently still properly references v2). With v3 I get

```
TypeError: Peaks.init(): The zoomview and/or overview container options must be valid HTML elements
    at checkContainerElements (peaks.esm.js:7794:12)
    at Peaks.init (peaks.esm.js:7821:9)
    at createPeaksInstance (AudioPeaks.vue:151:9)
    at AudioPeaks.vue:83:3
    at runtime-core.esm-bundler.js:2675:88
    at callWithErrorHandling (runtime-core.esm-bundler.js:158:18)
    at callWithAsyncErrorHandling (runtime-core.esm-bundler.js:166:17)
    at hook.__weh.hook.__weh (runtime-core.esm-bundler.js:2655:19)
    at flushPostFlushCbs (runtime-core.esm-bundler.js:325:26)
    at render2 (runtime-core.esm-bundler.js:6335:5)
```

I will update my component to work with peaks.js v3 in short order.

**Comment by suterma** [comment]
@newproplus and @chrisn  I have now upgraded to v3 and implemented the changes as suggested. Thanks for the heads up to me. You are invited to check the code as of now. I will release a new package later, after some other minor changes.

**Comment by chrisn** [comment]
Thanks! Sorry for the number of breaking changes...

**Comment by suterma** [comment]
@chrisn That's ok. Sometimes there is a need for it. I did not, however, gather the motivation for this current release from the docs.

**Comment by chrisn** [comment]
The main new features are overlay-style segments, the ability to drag segments, and insert segments by click+drag, and a scrollbar control. There's more detail in the [change log](https://github.com/bbc/peaks.js/blob/master/CHANGELOG.md).

**Comment by suterma** [comment]
I parsed through the changelog. Thanks for maintaining it, I observed some edges I need to look into also.

Hm, but what is the scroll bar control? I still only see the zoom and overview elements in https://waveform.prototyping.bbc.co.uk/ ? Am I supposed to add a third div somehow, as there is third option in getView()?

**Comment by chrisn** [comment]
I haven't updated that demo page yet ... have a look at https://github.com/bbc/peaks.js/blob/master/demo/scrollbar.html

![image](https://github.com/suterma/vue-peaks/assets/57790/294ae6f1-192c-4725-993f-79d9ee9016a6)

`getView()` works as before (it only returns the overview or zoomview waveforms), but you can use `getScrollbar()` to get the scrollbar object, e.g., if the scrollbar container element is resized: 
```js
peaks.views.getScrollbar().fitToContainer();
```

**Comment by suterma** [comment]
@chrisn @newproplus I have now released an updated package for vue-peaks (v2.0.0), I invite you to try it out, and report any remaining issues. It's still very lightweight and besides some default lazy loading, it's more or less exposing the same features as before (including direct access to the peaks.js instance)
